pkg/utils/coalescer.go

```diff
@@ -50,8 +50,12 @@ func Coalescer(interval time.Duration, input chan interface{}, stopCh <-chan str
 			}
 			// stop running the Coalescer only when all input+output channels are closed!
 			if !inputOpen {
+				// input is closed, so lets signal one last time if we have any pending unsignalled events
+				if signalled {
+					// send final event, so we dont miss the trailing event after input chan close
+					output <- struct{}{}
```

Comment:
The comment on `Coalescer`  is inaccurate
```after which
// it signals on output chan with the last value from input chan
```
and why remove closing the output channel?

Reply:
`Close` output channel  is better.